Region/RegionStats.cc

```diff
@@ -36,51 +37,81 @@ CARTA::SetHistogramRequirements_HistogramConfig RegionStats::getHistogramConfig(
     return config;
 }
 
-void RegionStats::getMinMax(float& minVal, float& maxVal, const std::vector<float>& data) {
-    // get min and max values in data
+// min max
+bool RegionStats::getMinMax(int channel, int stokes, float& minVal, float& maxVal) {
+    // Get stored min,max for given channel and stokes; return value indicates status
+    bool haveMinMax(false);
+    try {
+        minmax_t vals = m_minmax.at(stokes).at(channel);
+        minVal = vals.first;
+        maxVal = vals.second;
+        haveMinMax = true;
+    } catch (std::out_of_range) {
+        // not stored
+    }
+    return haveMinMax;
+}
+
+void RegionStats::setMinMax(int channel, int stokes, minmax_t minmaxVals) {
+    // Save min, max for given channel and stokes
+    if (channel == -2) { // all channels (cube); don't save intermediate channel min/max
```

Comment:
It would be good to use the `ALL_CHANNELS` constant for this consistently and avoid hardcoding the numeric value -- perhaps the constants should be defined in a different file which can be imported in multiple places. But it's not essential to fix this in this PR.